internal/pkg/aws/cloudformation/stackset/stackset.go

```diff
@@ -191,28 +191,45 @@ func (ss *StackSet) CreateInstancesAndWait(name string, accounts, regions []stri
 	return ss.WaitForOperation(name, id)
 }
 
+// InstanceSummary represents the identifiers for a stack instance.
+type InstanceSummary struct {
+	StackID string
+	Account string
+	Region  string
+	Status  InstanceStatus
+}
+
 // InstanceSummariesOption allows to filter instance summaries to retrieve for the stack set.
 type InstanceSummariesOption func(input *cloudformation.ListStackInstancesInput)
 
 // InstanceSummaries returns a list of unique identifiers for all the stack instances in a stack set.
 func (ss *StackSet) InstanceSummaries(name string, opts ...InstanceSummariesOption) ([]InstanceSummary, error) {
-	in := &cloudformation.ListStackInstancesInput{
-		StackSetName: aws.String(name),
-	}
-	for _, opt := range opts {
-		opt(in)
-	}
-	resp, err := ss.client.ListStackInstances(in)
-	if err != nil {
-		return nil, fmt.Errorf("list stack instances for stack set %s: %w", name, err)
-	}
+	var nextToken *string
 	var summaries []InstanceSummary
-	for _, summary := range resp.Summaries {
-		summaries = append(summaries, InstanceSummary{
-			StackID: aws.StringValue(summary.StackId),
-			Account: aws.StringValue(summary.Account),
-			Region:  aws.StringValue(summary.Region),
-		})
+	for {
+		in := &cloudformation.ListStackInstancesInput{
+			StackSetName: aws.String(name),
+			NextToken:    nextToken,
+		}
+		for _, opt := range opts {
+			opt(in)
+		}
```

Comment:
could just do this once outside of the for loop and then just update `in.NextToken`

Reply:
Done!